File: src/2529.c
#include <math.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int maximumCount(int *nums, int numsSize) {
    int retVal = 0;

    int negative = 0;
    int zero = 0;
    int positive = 0;
    int i;
    for (i = 0; i < numsSize; ++i) {
        if (nums[i] < 0) {
            ++negative;
        } else if (nums[i] == 0) {
            ++zero;
        } else {
            break;
        }
    }
    positive = numsSize - negative - zero;

    retVal = fmax(negative, positive);

    return retVal;
}

int main(int argc, char **argv) {
#define MAX_SIZE (8)
    struct testCaseType {
        int nums[MAX_SIZE];
        int numsSize;
    } testCase[] = {{{-2, -1, -1, 1, 2, 3}, 6}, {{-3, -2, -1, 0, 0, 1, 2}, 7}, {{5, 20, 66, 1314}, 4}};
    int numberOfTestCase = sizeof(testCase) / sizeof(testCase[0]);
    /* Example
     *  Input: nums = [-2,-1,-1,1,2,3]
     *  Output: 3
     *
     *  Input: nums = [-3,-2,-1,0,0,1,2]
     *  Output: 3
     *
     *  Input: nums = [5,20,66,1314]
     *  Output: 4
     */

    int answer = 0;
    int i, j;
    for (i = 0; i < numberOfTestCase; ++i) {
        printf("Input: nums = [");
        for (j = 0; j < testCase[i].numsSize; ++j) {
            printf("%s%d", (j == 0) ? "" : ",", testCase[i].nums[j]);
        }
        printf("]\n");

        answer = maximumCount(testCase[i].nums, testCase[i].numsSize);
        printf("Output: %d\n", answer);

        printf("\n");
    }

    return EXIT_SUCCESS;
}

File: src/2529.py
import sys
import logging
from typing import List, Optional
from collections import defaultdict, deque, Counter
from functools import cache, lru_cache
from bisect import bisect_left, bisect_right
from itertools import accumulate
from heapq import heapify, heappush, heappop


def logging_setting():
    LOG_LEVEL = logging.INFO    # DEBUG, INFO, WARNING, ERROR, CRITICAL
    if __debug__:
        LOG_LEVEL = logging.DEBUG

    logging.basicConfig(
        stream=sys.stderr,
        level=LOG_LEVEL,
        format="%(levelname)-6s %(asctime)s %(module)s:%(funcName)s:%(lineno)-4s %(message)s",
        datefmt='%Y/%m/%d %H:%M:%S')


class Solution:
    def maximumCount(self, nums: List[int]) -> int:
        retVal = 0

        numsSize = len(nums)

        negative = 0
        zero = 0
        positive = 0
        for i in range(numsSize):
            if nums[i] < 0:
                negative += 1
            elif nums[i] == 0:
                zero += 1
            else:
                break
        positive = numsSize - negative - zero

        retVal = max(negative, positive)

        return retVal


if __name__ == "__main__":
    logging_setting()

    try:
        logging.info("sys.version: %s", sys.version)
        print()

        pSolution = Solution()
        for nums in [[-2, -1, -1, 1, 2, 3], [-3, -2, -1, 0, 0, 1, 2], [5, 20, 66, 1314]]:
            # /* Example
            #  *  Input: nums = [-2,-1,-1,1,2,3]
            #  *  Output: 3
            #  *
            #  *  Input: nums = [-3,-2,-1,0,0,1,2]
            #  *  Output: 3
            #  *
            #  *  Input: nums = [5,20,66,1314]
            #  *  Output: 4
            #  */
            logging.info("Input: nums = %s", nums)

            retVal = pSolution.maximumCount(nums)
            logging.info("Output: %s", retVal)

            print()
    except KeyboardInterrupt as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass
    except Exception as exception:
        logging.error("%s: %s", exception.__class__.__name__, exception, exc_info=True)
        pass

    sys.exit(0)
